src/Utilities/PaymentUtils.res

```diff
@@ -237,3 +240,88 @@ let appendedCustomerAcceptance = (~isGuestCustomer, ~paymentType, ~body) => {
     ? body->Array.concat([("customer_acceptance", PaymentBody.customerAcceptanceBody)])
     : body
 }
+
+let usePaymentMethodTypeFromList = (~list, ~paymentMethod, ~paymentMethodType) => {
+  React.useMemo(() => {
+    PaymentMethodsRecord.getPaymentMethodTypeFromList(
+      ~list,
+      ~paymentMethod,
+      ~paymentMethodType=getPaymentMethodName(
+        ~paymentMethodType=paymentMethod,
+        ~paymentMethodName=paymentMethodType,
+      ),
+    )->Option.getOr(PaymentMethodsRecord.defaultPaymentMethodType)
+  }, (list, paymentMethod, paymentMethodType))
+}
+
+let useAreAllRequiredFieldsPrefilled = (~list, ~paymentMethod, ~paymentMethodType) => {
+  let paymentMethodTypes = usePaymentMethodTypeFromList(~list, ~paymentMethod, ~paymentMethodType)
+
+  paymentMethodTypes.required_fields->Array.reduce(true, (acc, requiredField) => {
+    acc && requiredField.value != ""
+  })
+}
+
+let useGetPaymentMethodList = (~list, ~paymentOptions, ~paymentType) => {
+  open Utils
+  let methodslist = Recoil.useRecoilValueFromAtom(RecoilAtoms.list)
+
+  let {showCardFormByDefault, paymentMethodOrder} = Recoil.useRecoilValueFromAtom(
+    RecoilAtoms.optionAtom,
+  )
+
+  let isApplePayReady = Recoil.useRecoilValueFromAtom(RecoilAtoms.isApplePayReady)
+  let isGooglePayReady = Recoil.useRecoilValueFromAtom(RecoilAtoms.isGooglePayReady)
+  let optionAtomValue = Recoil.useRecoilValueFromAtom(RecoilAtoms.optionAtom)
+
+  let paymentOrder = paymentMethodOrder->getOptionalArr->removeDuplicate
+
+  let areAllGooglePayRequiredFieldsPrefilled = useAreAllRequiredFieldsPrefilled(
+    ~list,
+    ~paymentMethod="wallet",
+    ~paymentMethodType="google_pay",
+  )
+
+  let areAllApplePayRequiredFieldsPrefilled = useAreAllRequiredFieldsPrefilled(
+    ~list,
+    ~paymentMethod="wallet",
+    ~paymentMethodType="apple_pay",
+  )
+
+  React.useMemo(() => {
+    switch methodslist {
+    | Loaded(paymentlist) =>
+      let paymentOrder =
+        paymentOrder->Array.length > 0 ? paymentOrder : PaymentModeType.defaultOrder
+      let plist = paymentlist->getDictFromJson->PaymentMethodsRecord.itemToObjMapper
+      let (wallets, otherOptions) =
+        plist->paymentListLookupNew(
+          ~order=paymentOrder,
+          ~showApplePay=isApplePayReady,
+          ~showGooglePay=isGooglePayReady,
+          ~areAllGooglePayRequiredFieldsPrefilled,
+          ~areAllApplePayRequiredFieldsPrefilled,
+          ~isShowPaypal=optionAtomValue.wallets.payPal === Auto,
```

Comment:
```suggestion
          ~showPaypal=optionAtomValue.wallets.payPal === Auto,
```

Reply:
I think we should keep the convention of putting is, has, before a variable which contains boolean